removed env vars requirement

imfractured opened the pull request:
found a way to remove the need for the source root ( #file ) 

added the first test for tapping the hello world and loading the data and checking for the string in the list

imfractured committed:
figured out a way to remove the need for the environment variables for accessing project root and setup the first dummy test

imfractured committed:
cleaning up a bit

MrMUsmanAli commented on demo.wiretapp/demo.wiretapp.xcodeproj/project.pbxproj:
```diff
@@ -459,7 +471,7 @@
 					"$(inherited)",
 					"@executable_path/Frameworks",
 				);
-				PRODUCT_BUNDLE_IDENTIFIER = "com.loblaw.fulfillment.mobilepicker.demo-wiretapp";
```
oh good catch! 🙏

MrMUsmanAli commented on demo.wiretapp/demo.wiretapp.xcodeproj/xcshareddata/xcschemes/demo.wiretapp.xcscheme:
```diff
@@ -70,18 +70,6 @@
             ReferencedContainer = "container:demo.wiretapp.xcodeproj">
          </BuildableReference>
       </BuildableProductRunnable>
-      <EnvironmentVariables>
```
✨

MrMUsmanAli commented on demo.wiretapp/demo.wiretapp/ContentView.swift:
```diff
@@ -1,10 +1,3 @@
-//
```
This reminds me that we'll need to setup linting

MrMUsmanAli replied:
I'll take this up today

MrMUsmanAli commented on demo.wiretapp/demo.wiretappUITests/WiretappBaseTestCase.swift:
```diff
@@ -0,0 +1,222 @@
+    }
+    
+    func getRootPath() -> String? {
+        let pathArray: [String] = #file.split(separator: "/")
```
Neat!

MrMUsmanAli replied:
I'm wondering.. does this mean the placement of this file has to always be 2 levels deep from the project root?

imfractured committed:
completed the implementation for the normal mock service

imfractured committed:
removed combine

imfractured committed:
scratching out the record structure

imfractured committed:
ops

imfractured committed:
threw the record code in to stop having to tab between projects

imfractured committed:
added an example custom url protocol that calls a urlsession internally and handles the client delegate methods in a different way

Muhammad-Usman-Ali committed:
- Plugged in the RecordURLSession implementation

- Removed the package file membership (Wiretapp, WiretappURLProtocol, WiretappSessionError) from the UITest target
- Moved conditionals to switch between Record and Mock modes into the canInit methods of the classes
- Temporarily added mock_responses environment variable to direct the recorded files

Muhammad-Usman-Ali committed:
- Fixed mock and record mode counter implementation

- Added support custom URLSession configuration

MrMUsmanAli committed:
Merge pull request #2 from MrMUsmanAli/feature/setup_record_mode

Plugged RecordURLSession into wiretapp
